### PR TITLE
Unique test meter names

### LINKED ISSUE #2533: Consider using the "fully" qualified test name to avoid conflicts
In the current unit test, `ActivitySource` and `Meter` names seem to be very ad-hoc. It could introduce confusion while troubleshooting, and what's worse - if test cases run in parallel, there could be conflicts.

Here are some examples:
* The class name is used as the Meter name (which means if there are multiple tests in the same class, the name clash would happen): https://github.com/open-telemetry/opentelemetry-dotnet/blob/0c93744d4ba51676b61ab1be38f7764190cfcec0/test/OpenTelemetry.Tests/Metrics/InMemoryExporterTests.cs#L29
* The method name and Meter name are inconsistent (probably a copypasta) https://github.com/open-telemetry/opentelemetry-dotnet/blob/0c93744d4ba51676b61ab1be38f7764190cfcec0/test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs#L76

The suggestion is to use a consistent paradigm https://github.com/open-telemetry/opentelemetry-dotnet/blob/0c93744d4ba51676b61ab1be38f7764190cfcec0/test/OpenTelemetry.Tests/Metrics/MemoryEfficiencyTests.cs#L29

### PR DESCRIPTION
**Pull request by mic-max**
Fixes #2533

## Changes

Make meter names unique in test suite
- Use `Utils.GetCurrentMethodName` for `Meter` constructors
  - Also if the test case includes parameters those should also be joined to end of the meter name separated with periods `.`
- Use `meter.Name` rather than using a matching string literal, requires `meter` to be initialized before call to `MeterProviderBuilder.AddMeter`
- Rename `meter1` to `meter` when only 1 `Meter` is created in the test function

All `Meter`'s currently being created are all added to a `MeterProvider` except a few (that were found manually by searching for references to the `Meter` constructor in filepaths containing the `test` substring) listed here by filepath, function name and variable name:
-`test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs`
  - `MeterSourcesWildcardSupportMatchTest`
    - `meter6`
  - `MeterSourcesWildcardSupportNegativeTestNoMeterAdded`
    - `meter1`
    - `meter2`
- `test\OpenTelemetry.Exporter.Prometheus.Tests\PrometheusExporterMiddlewareTests.cs`
  - `PrometheusExporterMiddlewareIntegration`
    - `meter`

Meter Names:
Manually gathered:
- `SomeCompany.SomeProduct.SomeComponent`
- `AbcCompany.XyzProduct.ComponentA.True`
- `AbcCompany.XyzProduct.ComponentA.False`
- `abcCompany.xYzProduct.componentC.True`
- `abcCompany.xYzProduct.componentC.False`
- `OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterMiddlewareTests..cctor`

Printed by debugger
```text
$ findstr "^Meter added:" .\meter.txt | sort

[Note: Start of wildcard naming test]
Meter Added: AbcCompany.XyzProduct.*
Meter Added: AbcCompany.XyzProduct.*
Meter Added: DefCompany.*.ComponentC
Meter Added: DefCompany.*.ComponentC
Meter Added: GhiCompany.qweProduct.ComponentN
Meter Added: GhiCompany.qweProduct.ComponentN
[Note: End of wildcard naming test]

Meter Added: OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests.OtlpMetricsExporterTests.ToOtlpResourceMetricsTest.False
Meter Added: OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests.OtlpMetricsExporterTests.ToOtlpResourceMetricsTest.True
Meter Added: OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterExtensionsTests.WriteMetricsCollectionTestWriteDoubleGauge
Meter Added: OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterExtensionsTests.WriteMetricsCollectionTestWriteDoubleSum
Meter Added: OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterExtensionsTests.WriteMetricsCollectionTestWriteHistogram
Meter Added: OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterExtensionsTests.WriteMetricsCollectionTestWriteLongGauge
Meter Added: OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterExtensionsTests.WriteMetricsCollectionTestWriteLongSum
Meter Added: OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterExtensionsTests.WriteMetricsCollectionTestWriteMultiple
Meter Added: OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterMetricsHttpServerTests+<PrometheusExporterMetricsHttpServerIntegration>d__0.MoveNext
Meter Added: OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterMiddlewareTests..cctor
Meter Added: OpenTelemetry.Instrumentation.AspNetCore
[Printed 21 times] Meter Added: OpenTelemetry.Instrumentation.Http
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.CounterAggregationTest.False
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.CounterAggregationTest.True
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.MultithreadedDoubleCounterTest
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.MultithreadedLongCounterTest
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.ObservableCounterAggregationTest.False
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.ObservableCounterAggregationTest.True
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.ObserverCallbackExceptionTest
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.ObserverCallbackTest
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.StreamNamesDuplicatesAreNotAllowedTest.1.Cumulative
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.StreamNamesDuplicatesAreNotAllowedTest.1.Delta
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.StreamNamesDuplicatesAreNotAllowedTest.2.Cumulative
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.StreamNamesDuplicatesAreNotAllowedTest.2.Delta
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.TestMetricPointCap.Cumulative
Meter Added: OpenTelemetry.Metrics.Tests.MetricApiTest.TestMetricPointCap.Delta
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToDropAndRetainInstrument
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToDropMultipleInstruments
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToDropSingleInstrument
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToProduceCustomHistogramBound
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToProduceMultipleStreamsFromInstrument
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToProduceMultipleStreamsWithDuplicatesFromInstrument
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToRenameMetric
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToRenameMetricConditionally1
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToRenameMetricConditionally2
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToRenameMetricWildCardMatch
Meter Added: OpenTelemetry.Metrics.Tests.MetricViewTests.ViewToSelectTagKeys
```

Aside: I added `Debug.WriteLine($"Meter Added: {name}");` to `src\OpenTelemetry\Metrics\MeterProviderBuilderBase.cs:AddMeter()` function, then selected all tests and ran in debug mode and copied the output from here to a text file
Visual Studio output window with "Show output from:" set to "Debug"

### CODE REVIEW

File: test/OpenTelemetry.Tests/Metrics/MemoryEfficiencyTests.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public class MemoryEfficiencyTests
         [InlineData(AggregationTemporality.Delta)]
         public void ExportOnlyWhenPointChanged(AggregationTemporality temporality)
         {
-            using var meter = new Meter(Utils.GetCurrentMethodName() + temporality, "1.0");
+            using var meter = new Meter(Utils.GetCurrentMethodName() + temporality);
```
Comment (reyang) [inline test/OpenTelemetry.Tests/Metrics/MemoryEfficiencyTests.cs:32]: Maybe adding a `.`? (or string interpolation)

```suggestion
            using var meter = new Meter(Utils.GetCurrentMethodName() + "." + temporality);
```

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterMiddlewareTests.cs [hunk 2]
```diff
@@ -26,13 +26,14 @@
...
 using OpenTelemetry.Metrics;
+using OpenTelemetry.Tests;
 using Xunit;
 
 namespace OpenTelemetry.Exporter.Prometheus.Tests
 {
     public sealed class PrometheusExporterMiddlewareTests
     {
-        private const string MeterName = "PrometheusExporterMiddlewareTests.Meter";
+        private static readonly string MeterName = Utils.GetCurrentMethodName();
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterMiddlewareTests.cs:36]: Curious - what would be the actual value? 😄

**Review by reyang — APPROVED** [review]

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterMiddlewareTests.cs (hunk 2 above)
Comment (mic-max) [reply test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterMiddlewareTests.cs:36]: So am I 😁
I will update the description with a list of all the meter names once I have printed them out to ensure uniqueness

File: test/OpenTelemetry.Tests/Metrics/MemoryEfficiencyTests.cs (hunk 1 above)
Comment (mic-max) [reply test/OpenTelemetry.Tests/Metrics/MemoryEfficiencyTests.cs:32]: Sure!

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterMiddlewareTests.cs (hunk 2 above)
Comment (mic-max) [reply test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterMiddlewareTests.cs:36]: The answer is `OpenTelemetry.Exporter.Prometheus.Tests.PrometheusExporterMiddlewareTests..cctor`

Comment (reyang) [reply test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterMiddlewareTests.cs:36]: That's good to know, thanks @mic-max!

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
@mic-max Please resolve the conflict and we are good to merge.

**Pull request merged**